From: Guillaume Brochu <guillaume.brochu@gmail.com>
Date: Fri, 30 Oct 2020 18:56:04 -0400
Subject: Makefile.debian

1. Add differentiated Makefile.debian files for optimized and non
optimised compilation. The appropriate method is selected in debian/rules

2. Remove "-warn-error A" to avoid future FTBFS on warnings.
Example : #521674

---
 tools/Makefile.debian.opt | 2 ++
 tools/Makefile.debian.out | 2 ++
 tools/Makefile.inc.unix   | 3 +--
 3 files changed, 5 insertions(+), 2 deletions(-)
 create mode 100644 tools/Makefile.debian.opt
 create mode 100644 tools/Makefile.debian.out

diff --git a/tools/Makefile.debian.opt b/tools/Makefile.debian.opt
new file mode 100644
index 0000000..68c7894
--- /dev/null
+++ b/tools/Makefile.debian.opt
@@ -0,0 +1,2 @@
+OCAMLC=ocamlc.opt
+OCAMLOPT=ocamlopt.opt
diff --git a/tools/Makefile.debian.out b/tools/Makefile.debian.out
new file mode 100644
index 0000000..9b9dfe4
--- /dev/null
+++ b/tools/Makefile.debian.out
@@ -0,0 +1,2 @@
+OCAMLC=ocamlc
+OCAMLOPT=ocamlopt
diff --git a/tools/Makefile.inc.unix b/tools/Makefile.inc.unix
index 676eb99..ba53046 100644
--- a/tools/Makefile.inc.unix
+++ b/tools/Makefile.inc.unix
@@ -6,8 +6,7 @@ CAMLP5_COMM=TOP=$(TOP) $(TOP)/tools/camlp5_comm.sh
 CAMLP5F=-DUNIX
 CAMLP5D=+camlp5
 LIBUNIX=-cclib -lunix
-OCAMLC=ocamlc.opt -w -3 -warn-error +a-52 -safe-string
-OCAMLOPT=ocamlopt.opt -w -3 -warn-error +a-52 -safe-string
+include $(GW_DIR)/tools/Makefile.debian
 OCAMLI=
 OCAMLD=
 STRIP=strip
